### DISCUSSION TITLE
Replace Python scripts with mason cli

### DISCUSSION THREAD

**Discussion by manh-t** [nimblehq/flutter_templates#116] [RFCs]
## Issue

The template was generated by Python, which is not a familiar programming language for our developers. Therefore, it could take effort when solving a problem related to the Python script.

## Solution

Now we have a new tool [mason cli](https://pub.dev/packages/mason_cli) that could help us generate the template quickly with less code. And we can take advantage of this tool to scale up the ability of the flutter_template, such as adding the optional features. Moreover, we can develop the brick in Dart, so everyone can use it.

## Who Benefits?

Flutter developers.

**Comment by sleepylee** [comment]
what are all the pros and cons of mason that you are aware/anticipated at this stage?

**Reply by manh-t** [reply]
@sleepylee For now, I can observe some pros and cons of `mason`:
Pros:
- Mason is written in Dart, and we can apply it to build the template for any project with any programming language.
- It’s possible to write a single piece of code as a brick template.
- The brick template can be applied and reused for existing projects.

Cons:
- When implementing complex features, the codebase might become messy because of the pattern of `mason`. For example, a file name now is `{{#snakeCase}}{{name}}{{/snakeCase}}_state.dart`, and we have to use this pattern for import as well.
- The brick template only supports adding optional features to a newly generated file, not overriding a file content in an existing project. Hence, it requires effort to add the feature to a project.

So, it is highly recommended to apply `mason` to the template level rather than an existing project 😄

**Comment by luongvo** [comment]
Let's close this RFC, we're working on a POC ticket [here](https://github.com/nimblehq/flutter_templates/issues/128).
